vm/src/cairo_run.rs

```diff
@@ -153,7 +159,13 @@ pub fn cairo_run_pie(
 
     cairo_runner
         .run_until_pc(end, &mut vm, hint_processor)
```

Comment:
I think I identified why this doesn't work. By this point the `Result<_, VirtualMachineError>` is already returned. What we need to make smaller is the `VirtualMachineError` itself.
Also, `ProgramError` is potentially big and contributes to `CairoRunError`'s size as well. This is because of the `IO` and `Parse` variants as well as all of the ones containing `String`.
It shouldn't matter for the common case as this happens once at the beginning and once at the end. However, `VirtualMachineError` is likely to be returned in several places.

Reply:
`RunnerError::PageNotOnSegment(Relocatable, usize)` is one candidate to boxing.

Reply:
`VirtualMachineError` is 32 bits already, should we make it even smaller?

Reply:
Bytes, but yeah, it's used in many places. I'm not sure the error is the culprit though.